### PR TITLE
[1.12.0][Bug][zos_mvs_raw] Bugfix/zos mvs raw verbose 1.12

### PR DESCRIPTION
**Pull request by fernandofloresg**
##### SUMMARY

Looks like mvccmd utility won't return anything in the stderr even if a program executed fails, so we only changed and if statement.

Changing that statement from 'and' to 'or' will in theory, fail any non-zero programs, which is better than having them fail only when verbose=True. Anyways, there is a task in the backlog to add a max_rc option to the module, so we can polish this mechanism there.

While testing this noticed that the module will hide the return code coming from the program if it fail, returning 8 as return code instead, fixed that and had to change some test cases that were expecting a wrong return code.

**NOTE: This was not originally added into the beta release because there was no time to test the fix with latest code. Good thing that we did, because we found and edge case. **

##### ISSUE TYPE

- Bugfix Pull Request
- 
##### COMPONENT NAME

zos_mvs_raw

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by fernandofloresg** [comment]
Screenshot 2024-11-01 at 4 31 13 PM
Screenshot 2024-11-01 at 4 30 57 PM

File: changelogs/fragments/1780-zos_mvs_raw-verbose-fail.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+bugfixes:
+  - zos_mvs_raw - If a program failed with a non-zero return code and verbose was false, the module would succeed.
```
Comment (ketankelkar) [inline changelogs/fragments/1780-zos_mvs_raw-verbose-fail.yml:2]:
```suggestion
  - zos_mvs_raw - If a program failed with a non-zero return code and verbose was false, the module would succeed (false positive).
```

i'm not sure if this suggested adds any clarity (keep it if it does 😄)

File: changelogs/fragments/1780-zos_mvs_raw-verbose-fail.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+    Whereas, if the program failed and verbose was true the module would fail.
```
Comment (ketankelkar) [inline changelogs/fragments/1780-zos_mvs_raw-verbose-fail.yml:3]: Does this sentence add anything? The module is supposed to fail when rc is nonzero, so this isn't describing a bug.
```suggestion
```

File: changelogs/fragments/1780-zos_mvs_raw-verbose-fail.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+    Fix now has a consistent behavior and fails in both cases.
```
Comment (ketankelkar) [inline changelogs/fragments/1780-zos_mvs_raw-verbose-fail.yml:4]: If we do end up removing the 'whereas...' sentence, this sentence here could be:
```suggestion
    Fix now fails the module for all instances where a program has a non-zero return code.
```

**Review by ketankelkar — CHANGES_REQUESTED** [review]
code changes look good, i had a few suggestions for the changelog, a few for comments in the changelog, and a potential mis-configured parameter. 

It's interesting that I've seen these changes 3 times now (from the other review) and only just now thought of suggestions here

File: changelogs/fragments/1780-zos_mvs_raw-verbose-fail.yml (hunk 1 above)
Comment (fernandofloresg) [reply changelogs/fragments/1780-zos_mvs_raw-verbose-fail.yml:2]: it does! thanks

**Comment by fernandofloresg** [comment]
@ketankelkar Thanks for your suggestions I ended up adding them all. Will run the test suite again and ask for a review after that is done.

**Review by richp405 — APPROVED** [review]
We have a similar issue in other functions, where we ended up letting the user select an 'acceptable' error code (I believe it was in submit).  What you did makes sense for mvs_raw.

**Comment by fernandofloresg** [comment]
Screenshot 2024-11-04 at 11 18 39 AM

**Comment by fernandofloresg** [comment]
closed it by accident

**Review by ddimatos — APPROVED** [review]
Looks good to me, looks like you found a legacy bug and fixed a new one and went with neither `and` and `or`. Thanks a bunch Fernando.

**Review by ketankelkar — APPROVED** [review]
LGTM - thanks for making the requested changes!
